system_modules/napmath/src/mathutils.h

```diff
@@ -388,6 +411,28 @@ namespace nap
 			return x * x * (3 - 2 * x);
 		}
 
+		template<typename T>
+		T smoothDamp(T current, T target, T& velocity, float deltaTime, float smoothTime, float maxSpeed)
+		{
+			const T dt = math::max<T>(math::epsilon<T>(), deltaTime);
+			const T smooth_time = math::max<T>(math::epsilon<T>(), smoothTime);
+
+			auto omega = T(2.0) / smooth_time;
+			auto x = omega * dt;
+			auto exp = T(1.0) / (T(1.0) + x + T(0.48) * x * x + T(0.235) * x * x * x);
+
+			auto delta_value = current - target;
+			if (maxSpeed < math::max<float>())
+			{
+				auto delta_max = smoothTime * maxSpeed;
+				delta_value = math::clamp<T>(delta_value, -delta_max, T(delta_max));
+			}
+			auto vel = (velocity + omega * delta_value) * dt;
+			velocity = (velocity - omega * vel) * exp;
+			return target + (delta_value + vel) * exp;
```

Comment:
If I read / understand this correctly, overshooting is now a possibility? That could impact the behaviour of projects that rely on this?